plugins/MauticCitrixBundle/EventListener/FormSubscriber.php

```diff
@@ -0,0 +1,638 @@
+
+        $products = [];
+        /** @var \Mautic\FormBundle\Entity\Field $field */
+        foreach ($fields as $field) {
+            if ('plugin.citrix.select.'.$product === $field->getType()) {
+                if (0 === count($productlist)) {
+                    $productlist = CitrixHelper::getCitrixChoices($product);
+                }
+                $alias = $field->getAlias();
+                /** @var array $productIds */
+                $productIds = $post[$alias];
+                if (!(array) $productIds) {
+                    $productIds = [$productIds];
+                }
+                foreach ($productIds as $productId) {
+                    $products[] = [
+                        'fieldName'    => $alias,
+                        'productId'    => $productId,
+                        'productTitle' => array_key_exists(
+                            $productId,
+                            $productlist
+                        ) ? $productlist[$productId] : 'untitled',
+                    ];
+                }
+            }
+        }
+
+        // gotoassist screen sharing does not need a product
+        if ('assist' !== $product) {
+            // check if there are products in the actions
+            /** @var Action $action */
+            foreach ($actions as $action) {
+                if (0 === strpos($action->getType(), 'plugin.citrix.action')) {
+                    if (0 === count($productlist)) {
+                        $productlist = CitrixHelper::getCitrixChoices($product);
+                    }
+                    $actionProduct = preg_filter('/^.+\.([^\.]+)$/', '$1', $action->getType());
+                    if (!CitrixHelper::isAuthorized('Goto'.$actionProduct)) {
+                        continue;
+                    }
+                    $actionAction = preg_filter('/^.+\.([^\.]+\.[^\.]+)$/', '$1', $action->getType());
+                    $productId    = $action->getProperties()['product'];
+                    if (array_key_exists(
+                        $productId,
+                        $productlist
+                    )) {
+                        $products[] = [
+                            'fieldName'    => str_replace('.', '_', $actionAction),
+                            'productId'    => $productId,
+                            'productTitle' => $productlist[$productId],
+                        ];
+                    }
+                }
+            }
+        }
+
+        return $products;
+    }
+
+    /**
+     * @param Events\FormEvent $event
+     *
+     * @throws \Doctrine\ORM\ORMInvalidArgumentException
+     * @throws \Doctrine\ORM\OptimisticLockException
+     * @throws \InvalidArgumentException
+     */
+    public function onFormPreSave(Events\FormEvent $event)
+    {
+        $form   = $event->getForm();
+        $fields = $form->getFields()->getValues();
+
+        // Verify if the form is well configured
+        if (0 !== count($fields)) {
```

Comment:
Every time I try to save the form, this is appended to my form title even though I have the email field mapped in the submit form action.

This feels a little hack (hijacking the entity title). Can we use exceptions to bubble this back up to the controller to set a FormError?